pkg/k8sapi/discovery.go

```diff
@@ -80,8 +81,21 @@ func NewController(clientset kubernetes.Interface) *Controller {
 }
 
 // CreateKubeClient creates a k8s client
-func CreateKubeClient() (clientset.Interface, error) {
-	kubeClient := k8sclient.GetKubeClient()
+func CreateKubeClient(kubeconfig, master string) (clientset.Interface, error) {
+	var kubeClient clientset.Interface
+	if master == "" && kubeconfig == "" {
+		kubeClient = k8sclient.GetKubeClient()
+	} else {
+		config, err := clientcmd.BuildConfigFromFlags(master, kubeconfig)
```

Comment:
Frankly, `client-go/tools/clientcmd.BuildConfigFromFlags` is able to accept empty kube API server URL and config path:

https://github.com/kubernetes/kubernetes/blob/1c974109b672ade9aae86f2daf1d50c89c36949f/staging/src/k8s.io/client-go/tools/clientcmd/client_config.go#L544

and if it does, loads the inClusterConfig appropriately, which is what the `k8sclient.GetKubeClient` call does. So, I would rather just remove the whole operator-sdk k8sclient pkg dependency here and just use client-go's clientcmd for this.

The dependency amazon-vpc-cni-k8s has on operator-sdk is ancient (0.0.7 which was released >1 year ago) and modern releases of operator-sdk don't even have a k8sclient package at all.